File: patterns/c-patterns/pattern123.c
// Pattern 123: ...

//   *
//   *
//  ***
//  ***
// *****
// *****

#include <stdio.h>

int main()
{
    int n = 3;
    int i, j, x, y, z = 1;

    for (i = 0; i < n; i++)
    {
        for (j = 2; j > 0; j--)
        {
            for (x = n - 1; x > i; x--)
            {
                printf(" ");
            }
            for (y = 0; y < z; y++)
            {
                printf("*");
            }
            printf("\n");
        }
        z += 2;
    }
    return 0;
}

File: patterns/javascript-patterns/pattern123.js
// Pattern 123: ...

//   *
//   *
//  ***
//  ***
// *****
// *****

const n = 3;
let z = 1;

for (let i = 0; i < n; i++) {
    for (let j = 2; j > 0; j--) {
        let spaces = '';
        for (let x = n - 1; x > i; x--) {
            spaces += ' ';
        }
        let stars = '';
        for (let y = 0; y < z; y++) {
            stars += '*';
        }
        console.log(spaces + stars);
    }
    z += 2;
}
